foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,674 @@
+
+        Returns the face stored in the mesh
+        identified by uuid. If such face do not
+        exists an exception is raised. Only CUBA.LABEL
+        data marking boundary faces is returned.
+
+        Parameters
+        ----------
+        uuid
+            uuid of the desired face.
+
+        Returns
+        -------
+        Face
+            Face identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the face identified by uuid was not found
+
+        """
+
+        try:
+            pointLabels = foamface.getFacePoints(self.name,
+                                                 self._uuidToFoamLabel[uuid])
+            puids = [self._foamPointLabelToUuid[lbl] for lbl in pointLabels]
+
+            # create patch data
+            patchNames = foamface.getBoundaryPatchNames(self.name)
+            patchFaces = foamface.getBoundaryPatchFaces(self.name)
+            i = 0
+            k = 0
+            facePatchMap = {}
+            while i < len(patchFaces):
+                start = i+1
+                end = start+patchFaces[i]
+                i += 1
+                for j in range(start, end):
+                    # here we assume that the boundaries are named
+                    # as boundary0,...
+                    # this to overcome limitation in tableextensio.pyx
+                    # at a moment
+                    facePatchMap[patchFaces[j]] = \
+                        int(patchNames[k].replace('boundary', ''))
+                    i += 1
+                k += 1
+
+            face = Face(puids, uuid)
+            if self._uuidToFoamLabel[uuid] in facePatchMap:
+                face.data[CUBA.LABEL] =\
+                    facePatchMap[self._uuidToFoamLabel[uuid]]
+            return face
+        except KeyError:
+            error_str = "Trying to get an non-existing edge with uuid: {}"
+            raise ValueError(error_str.format(uuid))
+
+    def get_cell(self, uuid):
+        """Returns a cell with a given uuid.
+
+        Returns the cell stored in the mesh
+        identified by uuid . If such cell do not
+        exists an exception is raised.
+
+        Parameters
+        ----------
+        uuid
+            uuid of the desired cell.
+
+        Returns
+        -------
+        Cell
+            Cell with id identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the cell identified by uuid was not found
+
+        """
+
+        try:
+
+            pointLabels = foamface.getCellPoints(self.name,
+                                                 self._uuidToFoamLabel[uuid])
+            puids = [self._foamPointLabelToUuid[lbl] for lbl in pointLabels]
+            cell = Cell(puids, uuid)
+
+            foamFiles = FoamFiles()
+            dataNames = foamFiles.get_cell_data_names(self.path,
+                                                      self._time)
+            for dataName in dataNames:
+                if dataName == "p":
```

Comment:
suggestion: There is a good amound of code duplication in this if..elif..else block. In this case you can use a python dictionary to simplify the code.
e.g:
```python
dataName2CUBA = {
    'p' : CUBA.PRESSURE.
    'p_rgh': CUBA.PRESSURE,
    'U': CUBA.VELOCITY,
    'alpha': CUBA.VOLUME_FRACTION,
    'phi': None,
    '': None
}
try:
    cuba = dataName2CUBA[dataName]
except KeyError
    error_str = "Data named "+dataName+" not supported"
    raise NotImplementedError(error_str)
else:
    if cuba is not None:
        cell.data[cuba] = foamFiles.get_cell_data(self.path, self._time, dataName, self._uuidToFoamLabel[uuid])

```

Please note that this is just suggestions and does not have to take place. However, I think that it makes the code easier to read and maintain especially when more dataNames are supported.